File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        
        l = -1
        r = len(nums) - 1
        
        while l + 1 < r:
            mid = (l + r) // 2
            
            if nums[mid] >= nums[0]:
                l = mid
            else:
                r = mid
                
        return min(nums[r], nums[0])

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMin = function(nums) {
    
    let l = -1
    let r = nums.length - 1
    
    while (l + 1 < r){
        mid = Math.floor( (l + r) / 2)
        
        if (nums[mid] >= nums[0]){
            l = mid
        }
        else{
            r = mid
        }
    }
    
    return Math.min(nums[r], nums[0])
    
};